WordPress/gutenberg
# Adds Metabox Support.

## Description
This adds metabox support with the UI @aduth came up with. See the accompanying markdown file in the docs.

## How Has This Been Tested?
Tested pretty extensively. Lots more polish work to do, but I believe this is merge-able-ish.
## Screenshots (jpeg or gifs if applicable):

## Types of changes
Adds PHP metabox support, through incredible hacks.

## Checklist:

## Testing Instructions:

- Verify that by default, on an install with no custom metaboxes, that no changes occur to the current Gutenberg experience.
- Verify that all core metaboxes do not appear, and no Extended Settings areas appear.
- Install ACF, Metabox.io, CMB2, or some other metabox creation framework, add their metaboxes to your theme, or generate some metaboxes of your own.
- Verify that metaboxes render in the Extended Settings area of your choice. There are normal, advanced, and side areas. advanced is currently merged into normal.
- Verify that normal metaboxes appear at the bottom of the editor area in an Editor Settings toggle panel.
- If you register sidebar metaboxes, verify that an 'Extended Settings' area appears in the sidebar.
- Verify that changing the form values in the metaboxes will allow you to update the post.
- Verify that returning the state of the form values to their original value will mark the metabox state as not dirty.
- When saving a post without changing metabox values, verify that they do not update.
- When saving a post with metabox changes, verify that the metaboxes indeed save the settings.
- If the sidebar only has changes and the normal area does not, verify that only the side bar saves, and vice versa.
- After successfully updating and verify the values saved correctly, try continual updates. The form data to dirty check against should now be the updated data.
